test/include/autopart_test.rb

```diff
@@ -0,0 +1,111 @@
+      end
+
+      def initialize
+        main
+      end
+      def planHasBoot
+        @planHasBoot
+      end
+      def planHasBoot=(v)
+        @planHasBoot = v
+      end     
+    end
+  end
+
+  subject(:client) { DummyYast::AutoinstallAutopartClient.new }
+
+  describe "#autopart" do
+
+    describe "ppc partitioning" do   
+      let(:target_map) { YAML.load_file(File.join(FIXTURES_PATH, 'storage', 'target_ppc.yml')) }
+      let(:ay_device) { Yast::Profile.current['partitioning'].find {|x| x["device"] == checked_device} }
+      before do
+        allow(Yast::Arch).to receive(:ppc).and_return(true)
+        allow(Yast::Partitions).to receive(:MinimalNeededBootsize).and_return(800000)
+        allow(Yast::Partitions).to receive(:DefaultBootFs).and_return(:ext4)
+        allow(Yast::Partitions).to receive(:FsidBoot).and_return(Yast::Partitions.fsid_gpt_prep)
+        Yast::Profile.ReadXML(File.join(FIXTURES_PATH, 'profiles', 'ppc_partitions.xml'))
+      end
+
+      context "AY configuration file has no /boot partition" do
+        before do
+          client.planHasBoot=false
+        end
+
+        context "checking device with root partition" do
+          let(:checked_device) { "/dev/sdc" }
+
+          it "do not add an additional prep disk if it is a pwoernv system" do
+            # bnc#989392
+            allow(Yast::Arch).to receive(:board_powernv).and_return(true)
+            ret = client.try_add_boot( ay_device, target_map[checked_device])
+            expect(ret["partitions"].size).to eq(ay_device["partitions"].size)
+          end
+          it "do add an additional prep disk if it is NOT a pwoernv system" do
+            allow(Yast::Arch).to receive(:board_powernv).and_return(false)
+            ret = client.try_add_boot( ay_device, target_map[checked_device])
+            expect(ret["partitions"].size).to eq(ay_device["partitions"].size+1)
+            # added disk has no mount point
+            expect((ret["partitions"] - ay_device["partitions"]).first.key?("mount")).to eq(false)
+            # added disk has no fsys
+            expect((ret["partitions"] - ay_device["partitions"]).first.key?("fsys")).to eq(false)
+          end
+        end
+
+        context "checking device with NO root partition" do
+          let(:checked_device) { "/dev/sdb" }
+
+          it "do not add an additional prep disk for ppc" do
+            allow(Yast::Arch).to receive(:board_powernv).and_return(false)
+            ret = client.try_add_boot( ay_device, target_map[checked_device])
+            expect(ret["partitions"].size).to eq(ay_device["partitions"].size)
+          end
+        end
+      end
+
+      context "AY configuration file has a /boot partition" do
+        before do
+          client.planHasBoot=true
+        end
+
+        context "checking device with root partition" do
+          let(:checked_device) { "/dev/sdc" }
+
+          it "do not add an additional prep disk if it is NOT a pwoernv system" do
```

Comment:
typo and I think here we should not mention powernv at all, as it is generic for all powerpcs.